File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/editor/controllers/SourceActionController.java
package ca.bc.gov.tno.areas.editor.controllers;

import java.util.List;

import javax.annotation.security.RolesAllowed;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import ca.bc.gov.tno.dal.db.services.interfaces.ISourceActionService;
import ca.bc.gov.tno.areas.editor.models.SourceActionModel;

/**
 * SourceActionController class, provides endpoints for actions.
 */
@RolesAllowed({ "administrator", "editor" })
@RestController("EditorSourceActionController")
@RequestMapping({ "/editor/source/actions", "/api/editor/source/actions" })
public class SourceActionController {

  /**
   * DAL for action.
   */
  private ISourceActionService actionService;

  /**
   * Creates a new instance of a SourceActionController object, initializes with
   * specified parameters.
   *
   * @param actionService SourceAction service.
   */
  @Autowired
  public SourceActionController(ISourceActionService actionService) {
    this.actionService = actionService;
  }

  /**
   * Request a list of all actions from the db.
   *
   * @return
   */
  @GetMapping(path = { "", "/" }, produces = MediaType.APPLICATION_JSON_VALUE)
  public List<SourceActionModel> findAll() {
    var results = actionService.findAll();
    var models = results.stream().map(a -> new SourceActionModel(a)).toList();
    return models;
  }

}

File: api/net/Areas/Editor/Controllers/SourceActionController.cs
using System.Net;
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;
using TNO.API.Areas.Editor.Models.SourceAction;
using TNO.API.Filters;
using TNO.API.Models;
using TNO.DAL.Services;
using TNO.Keycloak;

namespace TNO.API.Areas.Editor.Controllers;

/// <summary>
/// SourceActionController class, provides SourceAction endpoints for the api.
/// </summary>
[ClientRoleAuthorize(ClientRole.Editor)]
[ApiController]
[Area("editor")]
[ApiVersion("1.0")]
[Route("api/v{version:apiVersion}/[area]/source/actions")]
[Route("api/[area]/source/actions")]
[Route("v{version:apiVersion}/[area]/source/actions")]
[Route("[area]/source/actions")]
[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Unauthorized)]
[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
public class SourceActionController : ControllerBase
{
    #region Variables
    private readonly ISourceActionService _service;
    #endregion

    #region Constructors
    /// <summary>
    /// Creates a new instance of a SourceActionController object, initializes with specified parameters.
    /// </summary>
    /// <param name="service"></param>
    public SourceActionController(ISourceActionService service)
    {
        _service = service;
    }
    #endregion

    #region Endpoints
    /// <summary>
    /// Return an array of SourceSourceAction.
    /// </summary>
    /// <returns></returns>
    [HttpGet, HttpHead]
    [Produces("application/json")]
    [ProducesResponseType(typeof(IEnumerable<SourceActionModel>), (int)HttpStatusCode.OK)]
    [ProducesResponseType((int)HttpStatusCode.NotModified)]
    [SwaggerOperation(Tags = new[] { "SourceAction" })]
    [ETagCacheTableFilter("source_actions")]
    [ResponseCache(Duration = 5 * 60)]
    public IActionResult FindAll()
    {
        return new JsonResult(_service.FindAll().Select(c => new SourceActionModel(c)));
    }
    #endregion
}
